go/driver_test.go

```diff
@@ -2824,9 +2824,9 @@ func TestSnowflakeURIScheme(t *testing.T) {
 				require.Error(t, err, "Expected ParseDSN to fail for URI %q -> %q", tc.snowflakeURI, uri)
 
 				if tc.expectedErrCode != 0 {
-					sfError, ok := err.(*gosnowflake.SnowflakeError)
-					assert.True(t, ok, "Expected a SnowflakeError type for code %d", tc.expectedErrCode)
-					if ok {
+					var sfError *gosnowflake.SnowflakeError
+					assert.ErrorAs(t, err, &sfError, "Expected a SnowflakeError type for code %d", tc.expectedErrCode)
+					if sfError != nil {
```

Comment:
You can just omit this check; if sfError is nil, then the test has already failed. You can use `require.ErrorAs` instead if you want the test to end at that point.